flink-core/src/main/java/org/apache/flink/metrics/reporter/AbstractReporter.java

```diff
@@ -84,4 +84,24 @@ public void notifyOfRemovedMetric(Metric metric, String metricName, AbstractMetr
 	protected String replaceInvalidChars(String metricName) {
 		return metricName;
 	}
+
+	/**
+	 * Method which constructs the fully qualified metric name from the metric group and the metric
+	 * name.
+	 *
+	 * @param metricName Name of the metric
+	 * @param group Associated metric group
+	 * @return Fully qualified metric name
+	 */
+	private String constructMetricName(String metricName, AbstractMetricGroup group) {
+		StringBuilder builder = new StringBuilder();
+
+		for (String componentName : group.getScopeComponents()) {
+			builder.append(replaceInvalidChars(componentName)).append(".");
```

Comment:
this is a bit inefficient. The output of this loop is identical for all metrics on that group, yet is computed for every single metric. Instead you could modify the getScopeString() method to accept a charFilter argument that is passed into ScopeFormat.concat().